pkg/kubelet/pluginmanager/cache/actual_state_of_world.go

```diff
@@ -75,6 +75,18 @@ type actualStateOfWorld struct {
 
 var _ ActualStateOfWorld = &actualStateOfWorld{}
 
+// NamedPluginHandler holds information for handler and the Name of the plugin
+type NamedPluginHandler struct {
+	Handler PluginHandler
+	Name    string
+}
+
+// SocketPluginHandlers contains the map from socket path to NamedPluginHandler
+type SocketPluginHandlers struct {
+	Handlers map[string]NamedPluginHandler
+	sync.Mutex
+}
```

Comment:
Do we need import above two structs? Can we add plugin name and type info into the existing PluginInfo struct when the plugin finishes registering?

Reply:
PluginInfo is used by actualStateOfWorld struct :
```
	socketFileToInfo map[string]PluginInfo
```
For actualStateOfWorld, the above two structs are not relevant.
It seems we shouldn't conflate PluginInfo with such information.

Reply:
Why are they not relevant? Both plugin name and type are about plugin information, so why can't we add them into PluginInfo struct?

Reply:
In the new struct, type is not involved.
If we change the map to:
```
	Handlers map[string]PluginInfo
```
there would be redundancy - SocketPath is in both key and value.
It seems current formation is better.